web/ASC.Web.Client/src/components/pages/Settings/categories/common/team-template.js

```diff
@@ -0,0 +1,716 @@
+        !this.settingIsEqualInitialValue(settingName, valueFromSessionStorage)
+      ) {
+        isChanged = true;
+      }
+    });
+
+    if (!isChanged) {
+      this.setState({
+        id: customNames.id,
+        selectedOption: {
+          key: 0,
+          label: customNames.name,
+        },
+      });
+      saveToSessionStorage("selectedOption", customNames.name);
+    }
+
+    if (isChanged !== this.state.isChanged) {
+      this.setState({
+        isChanged: isChanged,
+      });
+    }
+  };
+
+  onChangeInput = (e) => {
+    const { teamTemplate } = this.props;
+    const { selectedOption } = this.state;
+
+    const name = e.target.name;
+    const value = e.target.value;
+
+    this.setState({ [name]: value });
+
+    if (selectedOption.label !== teamTemplate[3].name) {
+      this.setState({
+        id: teamTemplate[3].id,
+        selectedOption: {
+          key: 0,
+          label: teamTemplate[3].name,
+        },
+      });
+      saveToSessionStorage("selectedOption", teamTemplate[3].name);
+    }
+
+    if (this.settingIsEqualInitialValue(`${name}`, `${value}`)) {
+      saveToSessionStorage(`${name}`, "");
+    } else {
+      saveToSessionStorage(`${name}`, `${value}`);
+    }
+
+    if (value.length === 0) {
+      this.setState({
+        isChanged: true,
+      });
+    } else this.checkChanges();
+  };
+  isInvalidForm = () => {
+    const {
+      userCaption,
+      usersCaption,
+      groupCaption,
+      groupsCaption,
+      regDateCaption,
+      groupHeadCaption,
+      userPostCaption,
+      guestCaption,
+      guestsCaption,
+    } = this.state;
+
+    const errors = {
+      userCaption: !userCaption.trim(),
+      usersCaption: !usersCaption.trim(),
+      groupCaption: !groupCaption.trim(),
+      groupsCaption: !groupsCaption.trim(),
+      userPostCaption: !userPostCaption.trim(),
+      regDateCaption: !regDateCaption.trim(),
+      groupHeadCaption: !groupHeadCaption.trim(),
+      guestCaption: !guestCaption.trim(),
+      guestsCaption: !guestsCaption.trim(),
+    };
+
+    isError =
+      errors.userCaption ||
+      errors.usersCaption ||
+      errors.groupCaption ||
+      errors.groupsCaption ||
+      errors.userPostCaption ||
+      errors.regDateCaption ||
+      errors.groupHeadCaption ||
+      errors.guestCaption ||
+      errors.guestsCaption;
+
+    this.setState({ formErrors: errors });
+
+    return isError;
+  };
+
+  onSaveClick = () => {
+    const {
+      id,
+      userCaption,
+      usersCaption,
+      groupCaption,
+      groupsCaption,
+      regDateCaption,
+      groupHeadCaption,
+      userPostCaption,
+      guestCaption,
+      guestsCaption,
+      selectedOption,
+    } = this.state;
+
+    const {
+      setCurrentShema,
+      setCustomShema,
+      teamTemplate,
+      getCurrentCustomSchema,
+      t,
+      setIsLoading,
+    } = this.props;
+
+    if (this.isInvalidForm()) return;
+
+    if (selectedOption.label !== teamTemplate[3].name) {
```

Comment:
What is it teamTemplate[3]? Replace magic number

Reply:
Deleted teamTemplate[3]. Added customName and customId to compare with the currently selected option.